config/routes.rb

```diff
@@ -3,6 +3,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root 'users#index'
+
+  resources :users do 
+     resources :posts, only: [:index, :show]
+      post '/posts', to: 'posts#index'
```

Comment:
You don't need the custom post route on the `posts#index` . The post request for creating the post will be handled by `new` and `create` actions which you will add later in the module. Kindly remove this line of code.